diff -Naurp idn-1.2b/idn.c idn-1.2b.oden/idn.c
--- idn-1.2b/idn.c	2006-11-24 10:41:08.000000000 +0100
+++ idn-1.2b.oden/idn.c	2011-04-27 08:42:42.250866730 +0200
@@ -104,8 +104,10 @@ function_entry idn_functions[] = {
 
 	PHP_FE(idn_punycode_encode,				NULL)
 	PHP_FE(idn_punycode_decode,				NULL)
+#if ZEND_MODULE_API_NO < 20090626
 	PHP_FE(idn_to_ascii,					NULL)
 	PHP_FE(idn_to_utf8,						NULL)
+#endif
 	PHP_FE(idn_to_unicode,					NULL)
 
 #ifdef HAVE_IDN_TLD
@@ -676,6 +678,7 @@ PHP_FUNCTION(idn_punycode_decode)
 /*     IDNA wrappers     */
 /* --------------------- */
 
+#if ZEND_MODULE_API_NO < 20090626
 /* {{{ proto string idn_to_ascii(string input [, string charset])
    Convert to ACE according to IDNA
  */
@@ -697,7 +700,11 @@ PHP_FUNCTION(idn_to_ascii)
 	}
 
 	output = idn((*yyinput)->value.str.val, IDN_IDNA_TO_ASCII, charset);
-	RETVAL_STRINGL(output, strlen(output), 1);
+	if( output == NULL ) {
+		RETURN_NULL();
+	} else {
+		RETVAL_STRINGL(output, strlen(output), 1);
+	}
 	efree(output);
 }
 /* }}} */
@@ -723,10 +730,15 @@ PHP_FUNCTION(idn_to_utf8)
 	}
 
 	output = idn((*yyinput)->value.str.val, IDN_IDNA_TO_UNICODE, "UTF-8");
-	RETVAL_STRINGL(output, strlen(output), 1);
+	if( output == NULL ) {
+		RETURN_NULL();
+	} else {
+		RETVAL_STRINGL(output, strlen(output), 1);
+	}
 	efree(output);
 }
 /* }}} */
+#endif
 
 /* {{{ proto string idn_to_unicode(string input [, string charset])
    Convert from ACE according to IDNA
@@ -749,7 +761,11 @@ PHP_FUNCTION(idn_to_unicode)
 	}
 
 	output = idn((*yyinput)->value.str.val, IDN_IDNA_TO_UNICODE, charset);
-	RETVAL_STRINGL(output, strlen(output), 1);
+	if( output == NULL ) {
+		RETURN_NULL();
+	} else {
+		RETVAL_STRINGL(output, strlen(output), 1);
+	}
 	efree(output);
 }
 /* }}} */
diff -Naurp idn-1.2b/php_idn.h idn-1.2b.oden/php_idn.h
--- idn-1.2b/php_idn.h	2006-11-24 10:41:08.000000000 +0100
+++ idn-1.2b.oden/php_idn.h	2011-04-27 08:42:46.514891532 +0200
@@ -59,8 +59,10 @@ PHP_FUNCTION(idn_prep_iscsi);
 
 PHP_FUNCTION(idn_punycode_encode);
 PHP_FUNCTION(idn_punycode_decode);
+#if ZEND_MODULE_API_NO < 20090626
 PHP_FUNCTION(idn_to_ascii);
 PHP_FUNCTION(idn_to_utf8);
+#endif
 PHP_FUNCTION(idn_to_unicode);
 
 #ifdef HAVE_IDN_TLD
